engines/scumm/string.cpp

```diff
@@ -2075,4 +2075,30 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
 	memcpy(trans_buff, text, resStrLen(text) + 1);
 }
 
+Common::CodePage ScummEngine::getDialogCodePage(Common::String &message) const
+{
+	switch (_language) {
+	case Common::KO_KOR:
+		return Common::kWindows949;
+	case Common::JA_JPN:
+		return Common::kWindows932;
+	case Common::ZH_TWN:
+	case Common::ZH_CNA:
+		return Common::kWindows950;
+	case Common::RU_RUS:
+		return Common::kDos866;
+	case Common::HE_ISR:
+		switch (_game.id) {
+		case GID_LOOM:
+		case GID_ZAK:
+			fakeBidiString(reinterpret_cast<byte *>(message.begin()), true);
```

Comment:
I know this is super-ugly (especially with the current function name). If you have better ways, let me know :)

Reply:
No please don't modify variables inside a getter. Split this bit into a separate function

Reply:
Ok, separated.